core/src/main/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlanner.java

```diff
@@ -161,8 +161,11 @@ public void init(InitParameters params) {
 
       CompactionExecutorId ceid;
 
-      Objects.requireNonNull(executorConfig.type,
-          "'type' is a required and must be 'internal' or 'external'");
+      // If not supplied, GSON will leave type null. Default to internal
+      if (null == executorConfig.type) {
+        executorConfig.type = "internal";
+      }
```

Comment:
Not long ago, somebody swept through the code to standardize on `variable == null` rather than `null == variable`. It's not too important, but in order to not drift too far from that person's previous efforts to standardize things, we should probably strive to maintain that same style. I don't know if we can enforce this one in the style rules, though.

```suggestion
      // If not supplied, GSON will leave type null. Default to internal
      if (executorConfig.type == null) {
        executorConfig.type = "internal";
      }
```